src/qt/qvalidatedlineedit.cpp

```diff
@@ -106,6 +112,7 @@ void QValidatedLineEdit::checkValidity()
 void QValidatedLineEdit::setCheckValidator(const QValidator *v)
 {
     checkValidator = v;
+    checkValidity();
```

Comment:
This call might cause a side effect that is considered unwanted, no?

Here is an [analogous code:](https://code.woboq.org/qt5/qtbase/src/widgets/widgets/qlineedit.cpp.html#_ZN9QLineEdit12setValidatorEPK10QValidator):
```
void QLineEdit::setValidator(const QValidator *v)
{
    Q_D(QLineEdit);
    d->control->setValidator(v);
}
```
I'd leave `setCheckValidator()` function untouched.

Reply:
I don't know why it would be unwanted.

Reply:
b1a544be109d336c0b53722e3f8b51687972c94e

@hebasto don't understand what side effect you are talking about nor why you point to that code, `setValidator()` is not called.